lib/rubocop/cop/rspec/identical_equality_assertion.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for equality assertions with identical expressions on both sides.
+      #
+      # @example
+      #
+      #   # bad
+      #   expect(foo.bar).to eq(foo.bar)
+      #   expect(foo.bar).to eql(foo.bar)
+      #
+      #   # good
+      #   expect(foo.bar).to eq(2)
+      #   expect(foo.bar).to eql(2)
+      #
+      class IdenticalEqualityAssertion < Base
+        MSG = 'Identical equality can never fail.'
+
+        def_node_matcher :equality_check?, <<~PATTERN
+          (send (send nil? :expect $_) :to (send nil? {:eql :eq} $_))
```

Comment:
I suggest adding `be ==` and `be` here. This is optional though, `eq` will cover the majority of cases. Just afraid of people abusing alternative matches to make this pass.